arch/rv32i/src/pmp.rs

```diff
@@ -396,11 +396,10 @@ impl kernel::mpu::MPU for PMPConfig {
             initial_app_memory_size + initial_kernel_memory_size,
         );
 
-        let mut region_size =
-            math::PowerOfTwo::ceiling(memory_size as u32).as_num::<u32>() as usize;
-
         // RISC-V PMP is not inclusive of the final address, while Tock is, increase the memory_size by 1
-        region_size += 1;
+        let mut region_size = memory_size as usize + 1;
+
+        region_size = math::PowerOfTwo::ceiling(region_size as u32).as_num::<u32>() as usize;
```

Comment:
This is probably fine as it is, but can we make this into one expression to avoid having a `mut`?

I'm imagining something like this will work:
```
let region_size =
    math::PowerOfTwo::ceiling((memory_size + 1) as u32).as_num::<u32>() as usize;
```

Reply:
`region_size` is assigned later as well, so it still needs to be mut.

It's split out like this to be clearer, just to make it really obvious that we are adding 1 to the requested address, which at first isn't obvious.

Reply:
Fair enough. I guess everyone has their definition of cleaner, and this is purely cosmetic so I'm fine either way. :smile: